pkg/bpf/bpffs_migrate.go

```diff
@@ -119,6 +121,10 @@ func parseExtra(spec *ebpf.MapSpec, coll *ebpf.CollectionSpec) error {
 // compares its properties against the incoming spec and re-pins it to
 // ':pending' if any of its properties differ.
 func repinMap(bpffsPath string, name string, spec *ebpf.MapSpec) error {
+	if _, err := os.Stat(bpffsPath); errors.Is(err, fs.ErrNotExist) {
+		return fmt.Errorf("missing bpffs dir: %w", err)
+	}
```

Comment:
nit, Maybe a dumb question here, but the commit message says:

> The `tc/globals/` part of the path was missing, and the bpffs root was being scanned instead. Prevent this error from going unnoticed in the future by explicitly checking if the map directory exists.

If the directory path being passed in was `/sys/fs/bpf` instead of `/sys/fs/bpf/tc/globals`, then won't the directory check pass regardless of the bug? Also if we're part way through migrating these maps, then presumably the directories already exist somewhere, so why is it helpful to make these error checks here?

If there's a reasonable reason, then I'm fine with keeping these, my above questions are really just assessing what kind of difference they make and whether we need these at all (here and in `finalizeMap()`).

Reply:
I see this more as a courtesy to the user/caller, e.g. when `/sys/fs/bpf` is not mounted or `tc/globals` not present for some reason.

Actually, now I think of it, this might do more harm than good if migrations are run before iproute2 had the chance to create `tc/globals`. Missing pins are ignored (system already in desired state), but missing `tc/globals` is an error. I'll remove the checks, they don't really belong there.